denguefever_tw/dengue_linebot/views.py

```diff
@@ -316,3 +316,53 @@ def suggestion_list(request):
 def gov_report_list(request):
     context = {'gov_reports': GovReport.objects.all().order_by('-report_time')}
     return render(request, 'dengue_linebot/gov_report_list.html', context)
+
+
+@login_required
+def push_msg_form(request):
```

Comment:
Why not implement it through django froms?

https://docs.djangoproject.com/en/1.11/topics/forms/